src/System.Buffers/src/System/Buffers/ArrayPool.cs

```diff
@@ -37,17 +38,15 @@ public abstract class ArrayPool<T>
         /// </remarks>
         public static ArrayPool<T> Shared
         {
-            get
-            {
-                ArrayPool<T> instance = Volatile.Read(ref s_sharedInstance);
-                if (instance == null)
-                {
-                    Interlocked.CompareExchange(ref s_sharedInstance, Create(), null);
-                    instance = s_sharedInstance;
-                }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get { return Volatile.Read(ref s_sharedInstance) ?? EnsureSharedCreated(); }
+        }
 
-                return instance;
-            }
+        /// <summary>Ensures that <see cref="s_sharedInstance"/> has been initialized to a pool and returns it.</summary>
+        private static ArrayPool<T> EnsureSharedCreated()
+        {
+            Interlocked.CompareExchange(ref s_sharedInstance, Create(), null);
+            return s_sharedInstance;
```

Comment:
It's not, at least not on my machine, but I annotated EnsureSharedCreated anyway, just in case.